### PR TITLE
feat: serve ui from go server [WD-11656]

### PR DESCRIPTION
**Pull request by mas-who**
## Done

1. added api endpoints to serve ui static assets
2. updated ui build configs to include `/1.0/ui` as base prefix for routing

## Notes
1. The static assets will be copied from the ui directory and embedded in the go binary during compilation
2. multiple endpoints must be registered for various static file paths due to limitation from microcluster where api endpoints are registered individually behind `/1.0` prefix.
3. For development, the UI is served from vite dev server which can be reached on `http://0.0.0.0:8414/`. For testing and production environments (future), the UI is served from each cluster members API e.g. `https://0.0.0.0:9001/1.0/ui`

### CODE REVIEW

**Review by edlerd — COMMENTED** [review]
Code looks good, when I checkout and run make it fails atm. Shall we update the makefile to also run `cd ui && yarn build` and `cd api && go generate` for make to succeed on 1st run?

**Comment by mas-who** [comment]
> Code looks good, when I checkout and run make it fails atm. Shall we update the makefile to also run `cd ui && yarn build` and `cd api && go generate` for make to succeed on 1st run?

Aah good idea, will add that quick

File: Makefile [hunk 1]
```diff
@@ -4,6 +4,8 @@ default: build
 # Build targets.
 .PHONY: build
 build:
+	cd ui && yarn build
+	cp -r ui/build/ui api/static
```
Comment (edlerd) [inline Makefile]: Should this use `go:generate`?

**Review by edlerd — APPROVED** [review]
LGTM, just one question below.

File: Makefile (hunk 1 above)
Comment (mas-who) [reply Makefile]: 100%, sorry not sure how I missed that

**Pull request merged**